Src/EmbeddedScripts.JS.Jint/JintCodeRunner.cs

```diff
@@ -1,30 +1,55 @@
 ﻿using System;
 using System.Threading.Tasks;
 using EmbeddedScripts.Shared;
+using Esprima;
 using Jint;
+using Jint.Runtime;
 
 namespace EmbeddedScripts.JS.Jint
 {
     public class JintCodeRunner : ICodeRunner
     {
-        private Container container = new();
-        private Options jintOptions = new();
+        private Container _container = new();
+        private Options _jintOptions = new();
 
         public JintCodeRunner AddEngineOptions(Func<Options, Options> optionsFunc)
         {
-            jintOptions = optionsFunc(jintOptions);
+            _jintOptions = optionsFunc(_jintOptions);
             return this;
         }
         public ICodeRunner Register<T>(T obj, string alias)
         {
-            container.Register(obj, alias);
+            _container.Register(obj, alias);
             return this;
         }
 
-        public async Task RunAsync(string code) =>
-            await Task.Run(() => 
-                new Engine(jintOptions)
-                    .SetValuesFromContainer(container)
-                    .Execute(code));
+        public Task RunAsync(string code)
+        {
+            try
+            {
+                new Engine(_jintOptions)
+                    .SetValuesFromContainer(_container)
+                    .Execute(code);
+            }
+            catch (JavaScriptException e)
+            {
+                throw new ScriptRuntimeErrorException("Runtime error", e);
```

Comment:
looks like you reusing the same default texts across engines, so it could be a default arg values